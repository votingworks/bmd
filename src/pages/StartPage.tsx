import React, { useContext } from 'react'
import styled from 'styled-components'
import { RouteComponentProps, withRouter } from 'react-router-dom'

import BallotContext from '../contexts/ballotContext'

import { getPartyPrimaryAdjectiveFromBallotStyle } from '../utils/election'

import { Wobble } from '../components/Animations'
import ElectionInfo from '../components/ElectionInfo'
import LinkButton from '../components/LinkButton'
import Main, { MainChild } from '../components/Main'
import Prose from '../components/Prose'
import Sidebar from '../components/Sidebar'
import Screen from '../components/Screen'
import SettingsTextSize from '../components/SettingsTextSize'

const SidebarSpacer = styled.div`
  height: 90px;
`

type Props = RouteComponentProps<{}>

const StartPage = (props: Props) => {
  const { history } = props

  const {
    ballotStyleId,
    contests,
    election,
    precinctId,
    setUserSettings,
    userSettings,
    forceSaveVote,
  } = useContext(BallotContext)
  const { title } = election
  const partyPrimaryAdjective = getPartyPrimaryAdjectiveFromBallotStyle({
    election,
    ballotStyleId,
  })

  const onStart = () => {
    forceSaveVote()
    history.push('/contests/0')
  }

  return (
    <Screen>
      <Main>
        <MainChild center>
          <Prose textCenter>
            <h1 aria-label={`${partyPrimaryAdjective} ${title}.`}>
              {partyPrimaryAdjective} {title}
            </h1>
            <hr />
            <p>
              <span>
                Your ballot has <strong>{contests.length} contests</strong>.
              </span>
              <span className="screen-reader-only">
                When voting with the text-to-speech audio, use the accessible
                controller to navigate your ballot. To navigate through the
                contests, use the left and right buttons. To navigate through
                contest choices, use the up and down buttons. To select or
                unselect a contest choice as your vote, use the select button.
              </span>
            </p>
          </Prose>
        </MainChild>
      </Main>
      <Sidebar
        footer={
          <React.Fragment>
            <SettingsTextSize
              userSettings={userSettings}
              setUserSettings={setUserSettings}
            />
            <ElectionInfo
              election={election}
              ballotStyleId={ballotStyleId}
              precinctId={precinctId}
              horizontal
            />
          </React.Fragment>
        }
      >
        <Prose>
          <SidebarSpacer />
          <Wobble as="p">
            <LinkButton
              big
              primary
              onPress={onStart}
              id="next"
              aria-label="Press the right button to advance to the first contest."
            >
              Start Voting
            </LinkButton>
          </Wobble>
        </Prose>
      </Sidebar>
    </Screen>
  )
}

export default withRouter(StartPage)
