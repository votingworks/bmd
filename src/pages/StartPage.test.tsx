import React from 'react'
import { Route } from 'react-router-dom'

import { render } from '../../test/testUtils'
import electionSampleWithSeal from '../data/electionSampleWithSeal.json'
import electionSampleNoSeal from '../data/electionSampleNoSeal.json'
import electionPrimarySample from '../data/electionPrimarySample.json'

import StartPage from './StartPage'

it(`renders StartPage`, async () => {
  const { container, getByText } = render(
    <Route path="/" component={StartPage} />,
    {
      ballotStyleId: '12D',
      election: electionPrimarySample,
      precinctId: '23',
      route: '/',
    }
  )
  getByText('Democratic 2020 Primary Election')
  getByText(/Ballot Style: 12D/)
  expect(container.firstChild).toMatchSnapshot()
})

it(`renders StartPage with inline SVG`, async () => {
  const { container } = render(<Route path="/" component={StartPage} />, {
    election: electionSampleWithSeal,
    precinctId: '23',
    route: '/',
  })
  expect(container.firstChild).toMatchSnapshot()
})

it(`renders StartPage with no seal`, async () => {
  const { container } = render(<Route path="/" component={StartPage} />, {
    election: electionSampleNoSeal,
    precinctId: '23',
    route: '/',
  })
  expect(container.firstChild).toMatchSnapshot()
})
