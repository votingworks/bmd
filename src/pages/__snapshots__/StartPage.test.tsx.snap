// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders StartPage 1`] = `
.c4 {
  box-sizing: border-box;
  cursor: pointer;
  background: #4caf50;
  border: none;
  border-radius: 0.25rem;
  padding: 0.4rem 0.7rem;
  color: white;
  line-height: 1;
  white-space: nowrap;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: scroll;
  padding: 1rem 0.5rem 2rem;
}

.c1 {
  width: 100%;
  max-width: 35rem;
  margin: auto auto;
}

.c3 {
  line-height: 1.3;
  max-width: 66ch;
  text-align: center;
  margin: auto;
}

.c3 h1 {
  line-height: 1.1;
  margin: 2rem 0 1rem;
}

.c3 h2 {
  margin: 1.5rem 0 1rem;
}

.c3 h3,
.c3 p {
  margin: 1rem 0;
}

.c3 h1 + h2 {
  margin-top: -0.75rem;
}

.c3 h1 + p,
.c3 h2 + p,
.c3 h3 + p {
  margin-top: -0.75rem;
}

.c3 > :first-child {
  margin-top: 0;
}

.c3 > :last-child {
  margin-bottom: 0;
}

.c2 {
  max-width: 320px;
  margin: auto;
}

@media (min-width:480px) {
  .c0 {
    padding: 2rem 1rem;
  }
}

@media print {
  .c0 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    padding: 0;
  }
}

@media print {
  .c1 {
    max-width: 100%;
    margin: 0;
  }
}

<main
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <svg
        viewBox="0 0 500 500"
        xmlns="http://www.w3.org/2000/svg"
      >
        
  
        <circle
          cx="250"
          cy="250"
          fill="#fff"
          r="240"
          stroke="#000"
          stroke-width="10"
        />
        
  
        <text
          alignment-baseline="middle"
          style="font-family: sans-serif; font-size: 180%"
          text-anchor="middle"
          x="50%"
          y="50%"
        >
          Official Seal Here
        </text>
        

      </svg>
      

    </div>
    <div
      class="c3"
    >
      <h1>
        2020 General Election
      </h1>
      <p>
        Tuesday, November 6, 2020
        <br />
        Nearby County
        , 
        State of Somewhere
      </p>
      <p>
        <button
          class="c4"
          type="button"
        >
          Get Started
        </button>
      </p>
    </div>
  </div>
</main>
`;
